lib/base/process.cpp

```diff
@@ -917,7 +917,7 @@ void Process::Run(const std::function<void(const ProcessResult&)>& callback)
 		delete [] args;
 
 		if (callback)
-			Utility::QueueAsyncCallback(std::bind(callback, m_Result));
+			Utility::QueueAsyncCallback([this, &callback]() { return callback(m_Result); });
```

Comment:
Explain with your own words both (two) reasons why this will crash.

Reply:
I see only one reason here and that is I forgot to declare the return type forward.

Reply:
Are the things you use in the lambda captured by value or reference?

Reply:
Except the function callback all are  captured by value.

Reply:
Ok, _at least_ callback is captured by ref. Does this lambda run sync or async?

Reply:
According to the name of the method, it is an async callback. Would it be appropriate to make this ``this`` keyword reference counted?

Reply:
I think I don't even need to return a value here because ``callback`` is a void pointer.

Reply:
> Would it be appropriate to make this `this` keyword reference counted?

Despite you need only particular members you can include a ref counter as [here](https://github.com/Icinga/icinga2/blob/5e810f30a79ce7dcbf1520dfdf0635649cb54f71/lib/remote/jsonrpcconnection.cpp#L53-L58). But only if the used members in question won’t be modified anymore at this point (for the sake of thread-safety).